js/lib/credentials.js

```diff
@@ -0,0 +1,69 @@
+const grpc = require('grpc'),
+      request = require('request');
+
+class Credentials {
+  constructor(options={}) {
+    this.options = options;
+
+    let credentials = [];
+
+    if (this.options.ssl) {
+      credentials.push(grpc.credentials.createSsl());
+    }
+
+    credentials.push(grpc.credentials.createFromMetadataGenerator((...args) => {
+      return this.generate(...args);
+    }));
+
+    return grpc.credentials.combineChannelCredentials(...credentials);
+  }
+
+  generate(options, callback) {
+    this.getToken((err, data) => {
+      if (err) {
+        return callback({
+          code: err,
+          message: data
+        });
+      }
+
+      let metadata = new grpc.Metadata();
+
+      metadata.add('authorization', `Bearer ${data}`);
+      metadata.add('bluemix-instance', this.options.instance);
+
+      return callback(null, metadata);
+    });
+  }
+
+  getToken(callback) {
```

Comment:
Lets move this file to examples folder, but maybe keep the 

```
      metadata.add('authorization', `Bearer ${data}`);
      metadata.add('bluemix-instance', this.options.instance);
```

Reply:
And add hpcs connection address here

Reply:
Credentials is now part of the examples and we have a new `util.authMetadata(metadata, instance, token)` function to populate existing metadata objects for users.